explainer.md

```diff
@@ -0,0 +1,255 @@
+
+This feature provides developers with the same capability on the web, irrespective of whether the scenes in a transition are rendered across Documents.
+
+# Use-Cases
+A visual demo of the transition patterns targeted by this feature are [here](https://material.io/design/motion/the-motion-system.html#transition-patterns). The following is a summary of the semantics of these transition patterns :
+
+* Root Transitions : The full page content animates between 2 web pages with an optional static UI element on top. [Shared axis](https://material.io/design/motion/the-motion-system.html#shared-axis) shows an example.
+* Shared Element Transitions : A persistent UI element morphs into another (which could be a UI element on the next page or the whole page) changing its shape and content. [Container transform](https://material.io/design/motion/the-motion-system.html#container-transform) shows an example.
+* Entry/Exit Transitions : A UI element animates as it exits or enters the screen. This [issue](https://github.com/WICG/shared-element-transitions/issues/37) shows an example.
+
+These transitions should be feasible in SPAs (Single Page Apps) and MPAs (Multi Page Apps).
+
+# Design
+Let's take the example below which shows how the API can be used by a developer to animate the background and a shared element on a same origin navigation (MPA).
+
+### Old Document
+```
+<html>
+<head>
+  <style>
+    body {
+      background-color: blue;
+    }
+    .animated {
+      width: 100px;
+      height: 100px;
+      background-color: red;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" shared-id="shared-header" id="header"></div>
+</body>
+</html>
+```
+
+### New Document
+```
+<html>
+<head>
+  <style>
+    body {
+      background-color: lightblue;
+    }
+    .animated {
+      position: relative;
+      top: 100px;
+      width: 200px;
+      height: 200px;
+      background-color: red;
+    }
+    
+    @keyframes fadeIn {
+      0% {opacity: 0;}
+      100% {opacity: 1;}
+    }
+    
+    @keyframes fadeOut {
+      0% {opacity: 1;}
+      100% {opacity: 0;}
+    }
+    
+    ::shared-new(#root), ::shared-new(#shared-header) {
+      animation: fadeIn 1s;
+    }
+
+    ::shared-old(#root), ::shared-old(#shared-header) {
+      animation: fadeOut 1s;
+    }
+    
+    ::shared-container(#shared-header) {
+      transition: all 1s ease-in;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" shared-id="shared-header" id="header"></div>
+</body>
+</html>
+```
+
+The steps taken by the browser during the transition are as follows.
+
+1. When a navigation is initiated on the old Document, create the following pseudo elements in the top layer :
+    a. A container and child replaced element for each element with shared-id attribute. These are identified via ::shared-container(#shared-id) and ::shared-old(#shared-id) respectively.
```

Comment:
I'd verify the formatting, since in the rendered view these lines appear in the same line as above, which makes it look very confusing.

Also, do we need to explicitly call out that these pseudo would not have an associated ::backdrop (from https://fullscreen.spec.whatwg.org/#css-pe-backdrop), I'm guessing it's fine since pseudo elements can't have pseudo elements AFAIK.

Reply:
You're right that its probably obvious that a pseudo element won't have a pseudo element of its own but I added a footnote to mention this.